### PR TITLE
Remove Lombok usage in gateway-ha

### PR DESCRIPTION
**Pull request by mosabua**
- Refactor done with Delombok in Intellij, manuallly for simple ones,  and manual adjustments
- Removed unused loggers
- Changed Nullable and Nonnull annotations to consistently use jakarta annotations

Follow up to #104 and #105 

Part of #102

Note for reviewers: 

I kept formatting changes at minimum to allow relatively easy review. We will do a separate code reformmating to Trino/Airlift style in later PR (after existing PRs are hopefully in).

### CODE REVIEW

**Review by mosabua — COMMENTED** [review]
Still cleaning up and reducing the code formatting as much as possible to allow reasonably simple review.

**Review by mosabua — COMMENTED** [review]
All looking good now. Ready for review from others and hopefully not too much to review.

File: gateway-ha/src/main/java/io/trino/gateway/ha/config/AuthenticationConfiguration.java [hunk 1]
```diff
@@ -1,14 +1,91 @@
...
+    {
+        this.defaultType = defaultType;
+        this.oauth = oauth;
+        this.form = form;
+    }
+
+    public AuthenticationConfiguration() {}
+
+    public String getDefaultType()
+    {return this.defaultType;}
```
Comment (ebyhr) [inline gateway-ha/src/main/java/io/trino/gateway/ha/config/AuthenticationConfiguration.java:19]: This format is different from Airlift style. I would recommend formatting (Right click class -> Reformat Code) on IntelliJ after delombok.

Comment (mosabua) [reply gateway-ha/src/main/java/io/trino/gateway/ha/config/AuthenticationConfiguration.java:19]: I know .. all of the codebase does not follow airlift style yet .. the plan was to do a reformat after this PR or ideally after all the current PRs are merged. But given that you started the reformat already I think we should merge this PR .. and then do a reformat of that codebase in a follow up PR same as the other PR you sent.. and then get all other PRs rebased and such

**Comment by mosabua** [comment]
Please take a look @willmostly @vishalya and @Chaho12

**Review by ebyhr — APPROVED** [review]

**Pull request merged**